src/gui/accountstate.cpp

```diff
@@ -44,6 +44,7 @@ AccountState::AccountState(AccountPtr account)
     , _waitingForNewCredentials(false)
     , _maintenanceToConnectedDelay(60000 + (qrand() % (4 * 60000))) // 1-5min delay
     , _remoteWipe(new RemoteWipe(_account))
+    , _pushNotifications(QSharedPointer<PushNotifications>(new PushNotifications(account.data())))
```

Comment:
I wouldn't construct it unconditionally. Only if there's a valid URL for the websocket given in the capabilities.

Reply:
The idea behind this was: We always have the PushNotification object and then if someone wants to use push notifications it can check on the push notification object if any push notifications are available. With this we have the complete management of push notifications in the PushNotification object, so account state don't needs to care about push notifications.

Reply:
Yes, I get that but we create a few more objects per account + memory fragmentation for something which will in practice just sleep.

Reply:
okay. Just for my clarification: in the future, I guess, we will always use push notifications? So the push notification is always needed? It's just at the moment better/more efficient?

Reply:
Well, this needs to be deployed on the server and is an extra component. So even for future servers we can't assume it will be there, the sysadmins would need to put in the extra work to have said component.

That's why it's going through the capabilities system in a way, otherwise we'd just check the server version.